packages/headless/src/controllers/core/facets/facet-search/facet-search.ts

```diff
@@ -58,12 +60,20 @@ export function buildGenericFacetSearch<T extends FacetSearchState>(
           numberOfValues: options.numberOfValues + initialNumberOfValues,
         })
       );
-      dispatch(executeFacetSearch(facetId));
+      dispatch(
+        props.includeSearchContext
+          ? executeFacetSearch(facetId)
+          : executeFacetSearchWithoutSearchContext(facetId)
+      );
```

Comment:
Lots of plumbing required to pass down the intent not to include the search context...

Reply:
That's because the facet code has been extracted a few times 😄  No ways around it